services/blockchain-indexer/shared/dataService/business/token.js

```diff
@@ -86,30 +84,52 @@ const getTokenTopBalances = async (params) => {
 		data: {},
 		meta: {},
 	};
+
+	const { search, ...remParams } = params;
+
 	const accountBalancesTable = await getAccountBalancesTable();
-	const tokenInfos = await accountBalancesTable.find({ ...params }, ['address', 'balance']);
-
-	response.data[params.tokenID] = await BluebirdPromise.map(
-		tokenInfos,
-		async tokenInfo => {
-			const accountInfo = await getIndexedAccountInfo({ address: tokenInfo.address }, ['publicKey', 'name']);
-			const knowledge = getAccountKnowledge(tokenInfo.address);
-
-			return {
-				address: tokenInfo.address,
-				publicKey: accountInfo.publicKey,
-				name: accountInfo.name,
-				balance: tokenInfo.balance,
-				knowledge,
-			};
-		},
-		{ concurrency: tokenInfos.length },
-	);
+
+	remParams.leftOuterJoin = {
+		targetTable: 'accounts',
+		joinColumnLeft: 'account_balances.address',
+		joinColumnRight: 'accounts.address',
+	};
```

Comment:
we are not mentioning table name for input params (tokenID). could we get error for cases where both table has the same filtering column?

Reply:
Yes, if both tables have common column then we should query with `tableName.columnName`. But since `tokenID` is not present in accounts table we can just query with `tokenID`